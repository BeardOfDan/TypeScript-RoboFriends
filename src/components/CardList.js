import React from 'react';
import Card from './Card';

export default ({ robots }) => {
  return (
    <div className='tc'>
      {robots.map((robot, i, arr) => {
        return (
          <Card key={JSON.stringify(robot.username + i)} robot={robot} />
        );
      })}
    </div>
  );
};
