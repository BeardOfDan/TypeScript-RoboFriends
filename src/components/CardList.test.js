import { shallow } from 'enzyme';

import React from 'react';
import CardList from './CardList';

describe('CardList component', () => {
  it('should render', () => {
    const testRobots = [
      { name: 'firstName', email: 'firstEmail', id: 0 },
      { name: 'secondName', email: 'secondEmail', id: 1 },
      { name: 'thirdName', email: 'thirdEmail', id: 2 }
    ];
    expect(shallow(<CardList robots={testRobots} />).length).toEqual(1);
    expect(shallow(<CardList robots={testRobots} />)).toMatchSnapshot();
  });
});
