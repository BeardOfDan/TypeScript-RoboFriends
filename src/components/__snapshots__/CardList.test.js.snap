// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    robots={
        Array [
            Object {
              "email": "firstEmail",
              "id": 0,
              "name": "firstName",
            },
            Object {
              "email": "secondEmail",
              "id": 1,
              "name": "secondName",
            },
            Object {
              "email": "thirdEmail",
              "id": 2,
              "name": "thirdName",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          robot={
                    Object {
                              "email": "firstEmail",
                              "id": 0,
                              "name": "firstName",
                            }
          }
/>,
        <Unknown
          robot={
                    Object {
                              "email": "secondEmail",
                              "id": 1,
                              "name": "secondName",
                            }
          }
/>,
        <Unknown
          robot={
                    Object {
                              "email": "thirdEmail",
                              "id": 2,
                              "name": "thirdName",
                            }
          }
/>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "null",
        "nodeType": "function",
        "props": Object {
          "robot": Object {
            "email": "firstEmail",
            "id": 0,
            "name": "firstName",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "null",
        "nodeType": "function",
        "props": Object {
          "robot": Object {
            "email": "secondEmail",
            "id": 1,
            "name": "secondName",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "null",
        "nodeType": "function",
        "props": Object {
          "robot": Object {
            "email": "thirdEmail",
            "id": 2,
            "name": "thirdName",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            robot={
                        Object {
                                    "email": "firstEmail",
                                    "id": 0,
                                    "name": "firstName",
                                  }
            }
/>,
          <Unknown
            robot={
                        Object {
                                    "email": "secondEmail",
                                    "id": 1,
                                    "name": "secondName",
                                  }
            }
/>,
          <Unknown
            robot={
                        Object {
                                    "email": "thirdEmail",
                                    "id": 2,
                                    "name": "thirdName",
                                  }
            }
/>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "null",
          "nodeType": "function",
          "props": Object {
            "robot": Object {
              "email": "firstEmail",
              "id": 0,
              "name": "firstName",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "null",
          "nodeType": "function",
          "props": Object {
            "robot": Object {
              "email": "secondEmail",
              "id": 1,
              "name": "secondName",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "null",
          "nodeType": "function",
          "props": Object {
            "robot": Object {
              "email": "thirdEmail",
              "id": 2,
              "name": "thirdName",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
